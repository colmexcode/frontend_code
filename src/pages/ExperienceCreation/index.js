// ------------------------------ import libraries
import React from 'react';

// ------------------------------ import components
import { Hero } from '../../components/Hero';
import { ExperienceForm } from '../../components/ExperienceForm';
import { Footer } from '../../components/Footer';

// ------------------------------ import styles and images
// -------- import redux actions

// ------------------------------------ COMPONENT ------------------------------------//
// description of the component.
export default () => {
  return (
    <>
      <Hero />
      <ExperienceForm />
      <Footer />
    </>
  );
};
