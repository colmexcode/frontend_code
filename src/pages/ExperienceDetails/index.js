// ------------------------------ import libraries
import React, { useLayoutEffect } from 'react';
import { useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';

// ------------------------------ import components
import { Hero } from '../../components/Hero';
import { ExperienceHost } from '../../components/ExperienceHost';
import { Footer } from '../../components/Footer';
import heroPic from '../../assets/images/heroPicture.jpg';

// -------- import redux actions
import { setCurrentExperience } from '../../actions/experiencesActions';

// ------------------------------------ COMPONENT ------------------------------------//
// this is the page that shows the experiences details
// it is dynamic and depends of the given experience id

export default () => {
  const { id } = useParams();
  const dispatch = useDispatch();

  useLayoutEffect(() => {
    dispatch(setCurrentExperience(id));
  }, []);

  return (
    <>
      <Hero image={heroPic} />
      <ExperienceHost />
      <Footer />
    </>
  );
};
