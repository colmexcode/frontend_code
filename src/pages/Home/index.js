// ------------------------------ import libraries
import React from 'react';

// ------------------------------ import components
import { ExperienceCard } from '../../components/ExperienceCard';
import { Hero } from '../../components/Hero';
import { ExperiencesLayout } from '../../components/ExperiencesLayout';
import { Footer } from '../../components/Footer';
// ------------------------------ import styles and images

// ------------------------------------ COMPONENT ------------------------------------//
// this pages is the home.
// at first it has the top experiences according to the preferences of the user.
// user can search and find experiences by destinations.

export const HomePage = () => {
  return (
    <>
      <Hero />
      <ExperiencesLayout>
        <ExperienceCard />
      </ExperiencesLayout>
      <Footer />
    </>
  );
};
