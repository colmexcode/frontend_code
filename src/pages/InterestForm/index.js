// ------------------------------ import libraries
import React, { useState } from 'react';

// ------------------------------ import components
import { Button } from '../../global-styles/Buttons';

// ------------------------------ import styles and images
import {
  Container,
  Title,
  DescriptText,
  CategorySection,
  SelectionSection,
  TextChoice,
} from './styles';

// -------- import redux actions

// ------------------------------------ COMPONENT ------------------------------------//
// description of the component.
export default () => {
  const [categories, setCategories] = useState([]);
  return (
    <Container>
      <Title>WHAT ARE YOU UP TO?</Title>
      <DescriptText>
        For decades travellers have reached for Lonely Planet books
        when looking to plan and execute their perfect trip, but now,
        they can also let Lonely Planet Experiences lead the way.
      </DescriptText>
      <CategorySection>
        <Button
          name="Adventure Travel"
          onClick={(e) => setCategories(categories + e.target.name)}
        >
          Adventure Travel
        </Button>
        <Button
          name="Art And Culture"
          onClick={(e) => setCategories(categories + e.target.name)}
        >
          Art And Culture
        </Button>
        <Button
          name="Wildlife And Nature"
          onClick={(e) => setCategories(categories + e.target.name)}
        >
          Wildlife And Nature
        </Button>
        <Button
          name="Family Holidays"
          onClick={(e) => setCategories(categories + e.target.name)}
        >
          Family Holidays
        </Button>
        <Button
          name="Food And Drink"
          onClick={(e) => setCategories(categories + e.target.name)}
        >
          Food And Drink
        </Button>
      </CategorySection>
      <SelectionSection>
        <TextChoice>
          Awesome Choice:
          {categories}
        </TextChoice>
        <Button main>GO</Button>
      </SelectionSection>
    </Container>
  );
};
