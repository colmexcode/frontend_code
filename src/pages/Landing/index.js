// ------------------------------ import libraries
import React, { useRef } from 'react';

// ------------------------------ import components
import { Hero } from '../../components/Hero';
import { Map } from '../../components/Map';
import { LocationCard } from '../../components/LocationCard';
import { Footer } from '../../components/Footer';
import { TestimonialSection } from '../../components/TestimonialSection';
import { useGetMousePosition } from '../../hooks/useGetMousePosition';
import { LoginModal } from '../../components/LoginModal';
import { TrendingSection } from '../../components/TrendingSection';

// ------------------------------------ COMPONENT ------------------------------------//
// this is the landing page.
// This page has login or sign up buttons, the search bar, features location and testiomanials

export const LandingPage = () => {
  const mousePosition = useGetMousePosition();
  const modalRef = useRef();

  return (
    <>
      <Hero />
      <Map />
      <LocationCard top={mousePosition.y} left={mousePosition.x} />
      <TrendingSection />
      <TestimonialSection />
      <Footer />
      <LoginModal ref={modalRef} />
    </>
  );
};
