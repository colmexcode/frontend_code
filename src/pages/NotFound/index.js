// ------------------------------ import libraries
import React from 'react';

// ------------------------------ import components
import { Header } from '../../components/Header';
import { Footer } from '../../components/Footer';

// ------------------------------ import styles and images
import { Styled404, Section, Description } from './styles';
import { Button } from '../../global-styles/Buttons';

// ------------------------------------ COMPONENT ------------------------------------//
// this page is rendered when the route does not exist.

export default () => {
  return (
    <Styled404>
      <Header />
      <Section>
        <h1>404</h1>
        <Description>
          <p>Oops!</p>
          <p>Seems like you can't land here</p>
          <div>
            <Button main>Go Home</Button>
            <Button>About Us</Button>
          </div>
        </Description>
      </Section>
      {/* <Footer /> */}
    </Styled404>
  );
};
