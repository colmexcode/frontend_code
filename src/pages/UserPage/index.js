// ------------------------------ import libraries
import React from 'react';

// ------------------------------ import components
import { Hero } from '../../components/Hero';
import { UserProfile } from '../../components/UserProfile';
import { Footer } from '../../components/Footer';
import { UserExperiences } from '../../components/UserExperiences';
import { SectionSelector } from '../../components/SectionSelector';
import { ResulIndicator } from '../../components/ResultIndicator';

// ------------------------------------ COMPONENT ------------------------------------//
// this page has the user data and the experiences created and favorites.

export default () => {
  return (
    <>
      <ResulIndicator />
      <Hero />
      <UserProfile />
      <SectionSelector />
      <UserExperiences />
      <Footer />
    </>
  );
};
