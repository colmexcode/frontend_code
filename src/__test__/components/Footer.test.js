// ------------------------------ import libraries
import React from 'react';
import { mount } from 'enzyme';

// ------------------------------ import components
import { Home } from '../../pages/Home';

// ------------------------------------ TESTS ------------------------------------//
describe('<Home />', () => {
  test('Render Home component', () => {
    const Home = mount(<Home />);
    expect(Home.length).toEqual(1);
  });
});
