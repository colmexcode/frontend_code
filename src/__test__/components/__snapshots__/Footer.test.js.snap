// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Footer Snapshot 1`] = `
<footer
  className="styles__StyledFooter-sc-1569jkl-0 ipwbOo"
>
  <div
    className="styles__Description-sc-1569jkl-1 buvuhc"
  >
    <div
      className="styles__Rights-sc-1569jkl-2 bQlhYs"
    >
      <img
        alt="logo cozy place"
        src={Object {}}
      />
      <p>
        For decades travellers have reached for Lonely Planet books when looking to plan and execute their perfect trip, but now, they can also let Lonely Planet Experiences lead the way.
      </p>
      <p>
        LÂ©2020 Thousand Sunny. All rights reserved.
      </p>
    </div>
    <div
      className="styles__ListsGroup-sc-1569jkl-3 duvjPm"
    >
      <ul
        className="styles__List-sc-1569jkl-4 iDDVWk"
      >
        <li>
          destinations
        </li>
        <li>
          Mexico
        </li>
        <li>
          Colombia
        </li>
      </ul>
      <ul
        className="styles__List-sc-1569jkl-4 iDDVWk"
      >
        <li>
          interests
        </li>
        <li>
          Adventure Travel
        </li>
        <li>
          Art and Culture
        </li>
        <li>
          Wildlife and Nature
        </li>
        <li>
          Family Holidays
        </li>
        <li>
          Food and Drinks
        </li>
      </ul>
    </div>
  </div>
  <div>
    <input
      alt="facebook"
      className="styles__StyledIcon-urxmhz-0 djGORe"
      src={Object {}}
      type="image"
    />
    <input
      alt="instagram"
      className="styles__StyledIcon-urxmhz-0 djGORe"
      src={Object {}}
      type="image"
    />
    <input
      alt="youtube"
      className="styles__StyledIcon-urxmhz-0 djGORe"
      src={Object {}}
      type="image"
    />
    <input
      alt="twitter"
      className="styles__StyledIcon-urxmhz-0 djGORe"
      src={Object {}}
      type="image"
    />
  </div>
</footer>
`;
